src/serac/physics/base_physics.hpp

```diff
@@ -99,13 +92,91 @@ class BasePhysics {
    */
   virtual void completeSetup() = 0;
 
+  /**
+   * @brief Accessor for getting named finite element state fields from the physics modules
+   *
+   * @param state_name The name of the Finite Element State to retrieve
+   * @return The named Finite Element State
+   */
+  virtual const FiniteElementState& getState(const std::string& state_name) = 0;
+
+  /**
+   * @brief Get a vector of the finite element state solution variable names
+   *
+   * @return The solution variable names
+   */
+  virtual std::vector<std::string> getStateNames() = 0;
+
+  /**
+   * @brief Generate a finite element state object for the given parameter index
+   *
+   * @param parameter_index The index of the parameter to generate
+   * @param parameter_name The name of the parameter to generate
+   *
+   * @note The user is responsible for managing the lifetime of this object. It is required
+   * to exist whenever this physics module is called.
+   */
+  virtual std::unique_ptr<FiniteElementState> generateParameter(size_t             parameter_index,
```

Comment:
I think it would be more consistent with `setParameter()` to make the index the second argument